src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs

```diff
@@ -37,30 +39,94 @@ public static class HtmlHelperComponentExtensions
         /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
         /// <param name="parameters">An <see cref="object"/> containing the parameters to pass
         /// to the component.</param>
+        /// <param name="renderMode">The <see cref="RenderMode"/> for the component.</param>
         /// <returns>The HTML produced by the rendered <typeparamref name="TComponent"/>.</returns>
         public static async Task<IHtmlContent> RenderComponentAsync<TComponent>(
             this IHtmlHelper htmlHelper,
-            object parameters) where TComponent : IComponent
+            object parameters,
+            RenderMode renderMode) where TComponent : IComponent
         {
             if (htmlHelper == null)
             {
                 throw new ArgumentNullException(nameof(htmlHelper));
             }
 
-            var httpContext = htmlHelper.ViewContext.HttpContext;
-            var serviceProvider = httpContext.RequestServices;
-            var prerenderer = serviceProvider.GetRequiredService<StaticComponentRenderer>();
+            if (renderMode == default)
+            {
+                throw new ArgumentException("Can't render a component statically without prerendering it.", nameof(renderMode));
+            }
 
             var parametersCollection = parameters == null ?
                 ParameterView.Empty :
                 ParameterView.FromDictionary(HtmlHelper.ObjectToDictionary(parameters));
 
+            var context = htmlHelper.ViewContext.HttpContext;
+            switch (renderMode)
+            {
+                case RenderMode.Server:
+                    return NonPrerenderedBlazorComponent(context, typeof(TComponent), parametersCollection);
+                case RenderMode.ServerPrerendered:
+                    return await PrerenderedBlazorComponentAsync(context, typeof(TComponent), parametersCollection);
+                case RenderMode.Html:
```

Comment:
I'd really prefer this to be called static. Two of these options output html.

Reply:
Renamed to static